sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -169,16 +179,24 @@ object UnsupportedOperationChecker extends Logging {
           "DataFrames/Datasets")(plan)
     }
 
-    // Disallow multiple streaming aggregations
-    val aggregates = collectStreamingAggregates(plan)
+    val statefulOps = plan.collect {
+      case p: LogicalPlan if isStatefulOperation(p) => p
+    }
 
-    if (aggregates.size > 1 && outputMode != InternalOutputModes.Append) {
+    if (statefulOps.size > 1 &&
+      outputMode != InternalOutputModes.Append &&
+      SQLConf.get.statefulOperatorCorrectnessCheckEnabled) {
```

Comment:
I think this configuration should not just exist now as we unlocked various use cases (except stream-stream time interval join followed by other stateful operator), but let's handle it as separate issue. (I'll file a JIRA ticket on it.)

Reply:
Sure. IMHO it's still needed right now as we are still disabling 1) stream-stream time join as you mentioned above 2) multiple stateful ops when output mode is not append. 3) MapGroupWithState before aggregation. 

For 3). There are few checks added [as in this comment](https://github.com/apache/spark/pull/38503#discussion_r1016174482). Specifically, when `statefulOperatorCorrectnessCheckEnabled` is true, `MapGroupWithState->agg` is disallowed, when it is false, `MapGroupWithState->agg` is allowed to execute.